src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -671,7 +673,33 @@ private void TryObserveCollectionViewSource(object newValue)
 		// CollectionViewSource.View to whatever is being bound to (ie the ItemsSource). This should be fixed at some point because it
 		// has observable consequences in some usages.
 		internal object UnwrapItemsSource()
```

Comment:
this method is likely a hot path;
should we optimize it by:
- adding early-exit/fast-path for known types first?
- memorizing `IsGenericDescentOf` per `source.GetType()`?

or it matters little?